sdk/textanalytics/Azure.AI.TextAnalytics/tests/Infrastructure/TextAnalyticsClientLiveTestBase.cs

```diff
@@ -54,7 +57,9 @@ protected TextAnalyticsClient GetClient(AzureKeyCredential credential = default,
             }
             else
             {
-                credential ??= new AzureKeyCredential(TestEnvironment.ApiKey);
+                credential ??= useStaticResource
+                    ? new AzureKeyCredential(TestEnvironment.StaticApiKey)
+                    : new AzureKeyCredential(TestEnvironment.ApiKey);
```

Comment:
nitpick: could the conditional be moved inside the `AzureKeyCredential` constructor?

Reply:
The `AzureKeyCredential` class and its constructor are part of Azure Core. Since this dual need for static and dynamic test resources is very particular to Text Analytics, I was thinking that it is more appropriate to have this logic in the Text Analytics test infra.

Reply:
ops sorry for being unclear, I meant moved inside the constructor _call_, e.g. 
```csharp
credential ??= new AzureKeyCredential(useStaticResource ? TestEnvironment.StaticApiKey : TestEnvironment.ApiKey);
```